modules/rtm/R/twostream.R

```diff
@@ -0,0 +1,33 @@
+#' @name pro2s
+#'
+#' @title Coupled PROSPECT-Two-stream model
+#' @param param Model parameters, in the following order: N, Cab, (Car, Cbrown), Cw, Cm, solar zenith angle, LAI, soil_moisture
+#' @param prospect.version Version of PROSPECT to use (4, 5, or "5B"; default=5)
+
+pro2s <- function(param, prospect.version = 5){
+    prospect.version <- toupper(as.character(prospect.version))
+    plist <- as.list(param)
+    if (prospect.version == "4"){
+        if(length(plist) != 7) stop("Wrong number of parameters")
+        modname <- "pro42s"
+    }
+    else if (prospect.version == "5") {
+        if(length(plist) != 8) stop("Wrong number of parameters")
+        modname <- "pro52s"
+    }
+    else if (prospect.version == "5B") {
+        if(length(plist) != 9) stop("Wrong number of parameters")
+        modname <- "pro5b2s"
+    }
+    else stop("prospect.version must be 4, 5, or 5B") 
+    nw <- 2101
```

Comment:
It does need to be documented though, good catch.

In general, it would be nice to get around this, since the two-stream model should work on leaf optics of any dimensionality. I know Fortran can do arbitrary-dimension arrays, but I haven't been able to successfully implement them (i.e. have the two-stream subroutine intelligently get the dimensions of the input leaf optics and allocate all of the other arrays based on those dimensions). @serbinsh might have some thoughts on this too.